### ISSUE TITLE
libxkbcommon fails to compile

### ISSUE THREAD

**Issue by bertg42** [ni/nilrt#95]
Using the docker container, compiling libxkbcommon fails with this message:

| src/xkbcomp/parser.c: In function '_xkbcommon_parse':
| src/xkbcomp/parser.c:1839:12: error: 'YYEMPTY' undeclared (first use in this function); did you mean 'YYCOPY'?
|    yychar = YYEMPTY; /* Cause a token to be read.  */
|             ^~~~~~~
|             YYCOPY
| src/xkbcomp/parser.c:1839:12: note: each undeclared identifier is reported only once for each function it appears in
| src/xkbcomp/parser.c:1959:22: error: 'YYerror' undeclared (first use in this function); did you mean 'perror'?
|    else if (yychar == YYerror)
|                       ^~~~~~~
|                       perror
| src/xkbcomp/parser.c:1965:16: error: 'YYUNDEF' undeclared (first use in this function); did you mean 'YYENOMEM'?
|        yychar = YYUNDEF;
|                 ^~~~~~~
|                 YYENOMEM
| Makefile:1505: recipe for target 'src/xkbcomp/parser.lo' failed
| make[1]: *** [src/xkbcomp/parser.lo] Error 1
| make[1]: Leaving directory '/home/nilrt/nilrt/build/tmp-glibc/work/core2-64-nilrt-linux/libxkbcommon/0.8.0-r0/build'
| Makefile:1022: recipe for target 'all' failed
| make: *** [all] Error 2
| ERROR: oe_runmake failed
| WARNING: /home/nilrt/nilrt/build/tmp-glibc/work/core2-64-nilrt-linux/libxkbcommon/0.8.0-r0/temp/run.do_compile.2650918:1 exit 1 from 'exit 1'
| ERROR: Function failed: do_compile (log file is located at /home/nilrt/nilrt/build/tmp-glibc/work/core2-64-nilrt-linux/libxkbcommon/0.8.0-r0/temp/log.do_compile.2650918)
ERROR: Task (/home/nilrt/nilrt/sources/openembedded-core/meta/recipes-graphics/xorg-lib/libxkbcommon_0.8.0.bb:do_compile) failed with exit code '1'

**Comment by amstewart** [comment]
I can reproduce this error on my dev machine, so ACK on the failure.

But I don't see `libxkbcommon` anywhere in our package feeds. What target were you trying to build when you hit the failure?

**Comment by bertg42** [comment]
It was part of "bitbake --continue packagegroup-ni-extra", so I'm not totally sure which one.

**Comment by amstewart** [comment]
Ah. That's also how we build the extras feed internally. :) So, as long as you don't actually require the package for your project, you should be able to disregard any errors in that packagegroup.

**Comment by JKSH** [comment]
> I don't see `libxkbcommon` anywhere in our package feeds.

It has been available since v19.0: http://download.ni.com/ni-linux-rt/feeds/2019/x64/core2-64/libxkbcommon_0.8.0-r0.7_core2-64.ipk

I'm using it for a CompactRIO project (v20.5). Please don't take it away!

**Comment by amstewart** [comment]
After discussion offline, @bertg42 has decided that `libxkbcommon` is not needed for his project. Its presence in the 'extra' package feed means that it isn't on the roadmap for the NI RTOS team to support, so I'm moving to close the issue.

I've followed up with @JKSH offline to register his support request with the team's internal planners. If anyone else has a use-case which requires this package, please reach out to me directly and I'll connect you as well.
